### DISCUSSION TITLE
How to add elements to ui.expansion() header?

### DISCUSSION THREAD

**Discussion by HilkopterBob** [zauberzeug/nicegui#655] [Q&A] [answered]
Is it possible to add ui.components to the header of the ui.expansion element?

![image](https://user-images.githubusercontent.com/117028104/228254428-3c68c6f7-e4a4-418b-8da9-96f14e1db158.png)

**Comment by falkoschindler** [comment, marked as answer]
The `ui.expansion` provides the slots from [Quasar's QExpansionItem](https://quasar.dev/vue-components/expansion-item). You can fill it with UI elements like this:

```py
with ui.expansion('') as expansion:
    with expansion.add_slot('header'):
        ui.image('https://nicegui.io/logo.png').classes('w-16')
    ui.label('What a nice GUI!')
```

The `label` argument is non-optional, but ineffective in this case. Therefore it is simply set to an empty string.

**Reply by rodja** [reply]
@HilkopterBob this would be another nice demo for the documentation ;-)

**Reply by falkoschindler** [reply]
@rodja Sorry, too late. Once I had written the code snippet, I just added it to the documentation myself.
